### PR TITLE
Manage iOS contexts separately

### PR DESCRIPTION
**Pull request by gw280**
Add `#pragma clang diagnostic ignored "-Wobjc-method-access"` to ignore the warnings. This is because we're unit testing using non-public APIs (this was discussed and approved in https://github.com/flutter/engine/pull/11798)

### CODE REVIEW

**Comment by gw280** [comment]
Sorry I think the discussion surrounding private APIs for testing was discussed on Discord, not in the PR. In any case, we decided to go ahead with it.

File: testing/scenario_app/ios/Scenarios/ScenariosTests/FlutterViewTest.m [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+#import <Flutter/Flutter.h>
+#import <XCTest/XCTest.h>
+#import "AppDelegate.h"
+
+@interface FlutterViewGLContextTest : XCTestCase
+@property(nonatomic, strong) FlutterViewController* flutterViewController;
+@property(nonatomic, strong) FlutterEngine* flutterEngine;
+@end
+
+#pragma clang diagnostic push
```
Comment (dnfield) [inline testing/scenario_app/ios/Scenarios/ScenariosTests/FlutterViewTest.m]: can we move this to just around the offending area, so we don't open it up for the whole file?

**Comment by gw280** [comment]
> This sort of a change makes me really really nervous. We have never ignored warnings when they became inconvenient. As written, this code has a very real issue and I would really rather not characterize the fix as undertaking a yak shave if the workaround involves selectively disabling a warning.

So my understanding was that this was strictly temporary until we can figure out a better way to test this, hence the meeting tomorrow and the large TODO comment. I'd like for this to not be blocked on a test issue like this, ugliness notwithstanding.

**Comment by dnfield** [comment]
@chinmaygarde - this change is only within a unit test. 

Do you think it's ok to do this while we work on wiring up a better way to test these things?  Otherwise, we can't land the fix with a test, and we risk landing a fix that we will quickly regress because it has no tests.

**Comment by chinmaygarde** [comment]
> this change is only within a unit test.

Code in unit-tests is still code we have to maintain. The same engineering practices and standard ought to apply to all code.

I am convinced we should work on wiring up a harness ASAP. Will work on this now. George, feel free to land this (you have Dan's LGTM already) while I/we do this.

**Comment by gw280** [comment]
I found some more issues with this locally so I'm going to shelve this until we meet to discuss the testing changes needed to support this change.

**Pull request merged**